### PR TITLE
Update bazel test job to 0.4.2 and support //build-tools/... and //build/...

### PR DESCRIPTION
**Pull request by ixdy**
Necessary for https://github.com/kubernetes/kubernetes/pull/38736 to work.

x-ref https://github.com/kubernetes/kubernetes/issues/38126

### CODE REVIEW

**Comment by ixdy** [comment]
I'll rebase and fold the bazel update into this PR too.

**Comment by ixdy** [comment]
I've rebased `jobs/pull-kubernetes-bazel.sh` and I've updated the bazel image to bazel 0.4.2.

I haven't pushed any images or configs yet.

I don't know how to test this, either.

File: images/pull-kubernetes-bazel/Dockerfile [hunk 1]
```diff
@@ -29,9 +29,10 @@ RUN apt-get update && apt-get install -y \
     python && \
     apt-get clean
 
-RUN wget https://github.com/bazelbuild/bazel/releases/download/0.3.2/bazel_0.3.2-linux-x86_64.deb && \
-    dpkg -i bazel_0.3.2-linux-x86_64.deb && \
-    rm bazel_0.3.2-linux-x86_64.deb
+ENV BAZEL_VERSION 0.4.2
+RUN wget "https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel_${BAZEL_VERSION}-linux-x86_64.deb" && \
```
Comment (spxtr) [inline images/pull-kubernetes-bazel/Dockerfile]: Does it make sense to do `RUN BAZEL_VERSION=0.4.2 wget ...` instead?

File: jobs/pull-kubernetes-bazel.sh [hunk 2]
```diff
@@ -28,8 +28,13 @@ release-1.0|release-1.1|release-1.2|release-1.3|release-1.4|release-1.5)
   ;;
 esac
 
+build_dir=build
```
Comment (spxtr) [inline jobs/pull-kubernetes-bazel.sh]: TODO: remove this block?

**Comment by spxtr** [comment]
I don't think we have a way to test job changes in other repos as a presubmit to this repo. That's certainly something we should add, though.

File: jobs/pull-kubernetes-bazel.sh (hunk 2 above)
Comment (ixdy) [inline jobs/pull-kubernetes-bazel.sh]: done

**Comment by spxtr** [comment]
/lgtm

be sure to push the new image before updating the job config :)

**Comment by ixdy** [comment]
Yep, `bazelbuild:0.1` has been pushed.

**Pull request merged**